### ISSUE TITLE
A multiremote browser does everything at the same time

### ISSUE THREAD

**Issue by kajmagnus** [webdriverio/webdriverio#1548]
## The problem

(There's a minimal example a bit below.)

I'm testing a chat app, like Gitter and Slack, in two browsers. However, a multiremote Webdriver browser instance, sends all commands to Chrome at the same time. Look here:

```
        browserA.click('.esTB_SearchBtn');
        browserA.waitForVisible('input[name="searchPhrase"]');
        browserA.waitForEnabled('input[name="searchPhrase"]');
        browserA.setValue('input[name="searchPhrase"]', '134567890abcdefghijklmnopqrstuvwxyz');
        browserA.click('button[type="submit"]');

```

all these commands get executed by Webdriver at the same time (i.e. without waiting for the previous steps to complete). The end result is that the browser does everything in the wrong order: attempts to type in the search field, before it's visible. Attempts to click the submit button, before it has been created.

However, Webdriver.io should do these things one at a time, not continuing on the next line, before the current line has been done & completed.

*Without* multiremote, this works as expected. (Example below.)

## Environment

* WebdriverIO version: `v4.2.5`
* Node.js version: `node -v` –> `v5.0.0`
* [Standalone mode or wdio testrunner](http://webdriver.io/guide/getstarted/modes.html): `wdio` testrunner
* if wdio testrunner, running synchronous or asynchronous tests: "sync mode is enabled by default when using the wdio testrunner" (wrote Christian Boman in the Gitter chat)
* Additional wdio packages used (if applicable): None, as far as I know. See example code below.

## Link to Selenium/WebdriverIO logs

A [gist](https://gist.github.com) with the Selenium/WebdriverIO logs:

https://gist.github.com/kajmagnus/bf5f5a41086ab8029727c95a986c36e8

## Code To Reproduce Issue

In a file `two-chrome.conf.js`:
```
var wdioConf = { config: {
    capabilities: {
        browserA: { desiredCapabilities: { browserName: 'chrome' } },
        browserB: { desiredCapabilities: { browserName: 'chrome' } }
    },

    // This also won't work:
    // capabilities: { browserA: { desiredCapabilities: { browserName: 'chrome' } } },

    // This works fine:
    //  capabilities: [{ browserName: 'chrome' }],

    logLevel: 'error',
    coloredLogs: true,
    screenshotPath: './target/e2e-test-error-shots/',
    waitforTimeout: 999999,
    connectionRetryTimeout: 999999,
    connectionRetryCount: 3,
    framework: 'mocha',
    reporters: ['dot'],
    mochaOpts: {
        ui: 'bdd',
        timeout: 99999999
    }
} };

module.exports = wdioConf;
```

And a file `multiremote-please-wait.test.js`:
```
describe('multiremote', function () {
    it('should do one thing at a time', function () {
        if (typeof browserA === 'undefined')
            browserA = browser;  // if just 1 browser

        browserA.url('https://www.effectivediscussions.org/dev/-81n25/technical-information');
        browser.debug();
        browserA.click('.esTB_SearchBtn');
        browserA.waitForVisible('input[name="searchPhrase"]');
        browserA.waitForEnabled('input[name="searchPhrase"]');
        //browser.debug();    <-- comment in, and this test will work (i.e. submit the search form)
        browserA.setValue('input[name="searchPhrase"]', '134567890abcdefghijklmnopqrstuvwxyz');
        //browser.debug();    <-- this one too
        browserA.click('button[type="submit"]');
        browser.debug();
    });
});
```

Create the files, and run the test, like so: (note that there's a `debug()` so you need to hit Enter)
```
node_modules/webdriverio/bin/wdio two-chrome.conf.js --spec multiremote-please-wait.test.js
```

The test fails: nothing gets typed in the search form, and it won't get submitted. However, comment in the `browser.debug()` lines above, and it'll work.

Or change the capabilities from 2 browsers to just 1, by using this instead:
`capabilities: [{ browserName: 'chrome' }],`  (commented out above)
 — then the test will work fine, too (without the extra `debug()`).

As you can see in a comment above, this:
`capabilities: { browserA: { desiredCapabilities: { browserName: 'chrome' } } },`
also results in the same everything-at-the-same-time problem — although just one browser.

Therefore I'm wondering if this way of configuring things:
`capabilities: { browserA: { desiredCapabilities: { browserName: 'chrome' } } },`
somehow _disables_ sync mode.

But this enables it:  `capabilities: [{ browserName: 'chrome' }],`

**Comment by christian-bromann** [comment]
I don't know why it is not working for you but I just executed
```js
describe('multiremote', function () {
  it('should do one thing at a time', function () {
    browser.url('https://www.effectivediscussions.org/dev/-81n25/technical-information');
    browser.click('.esTB_SearchBtn');
    browser.waitForVisible('input[name="searchPhrase"]');
    browser.waitForEnabled('input[name="searchPhrase"]');
    browser.setValue('input[name="searchPhrase"]', '134567890abcdefghijklmnopqrstuvwxyz');
    browser.click('button[type="submit"]');
  });
});
```
and both browser did that flawlessly. However in your logs I see this:
```js
08:47:16.500 INFO - Executing: [find element: By.cssSelector: .esTB_SearchBtn])
08:47:16.500 INFO - Executing: [find elements: By.cssSelector: input[name="searchPhrase"]])
08:47:16.500 INFO - Executing: [find elements: By.cssSelector: input[name="searchPhrase"]])
08:47:16.500 INFO - Executing: [find element: By.cssSelector: button[type="submit"]])
08:47:16.500 INFO - Executing: [find elements: By.cssSelector: input[name="searchPhrase"]])
```
and your execution time is pretty low (mine is 12.80s without the debugs). Maybe start with a real simple example and work the way up. I can't spot anything wrong with the code. However it's working fine for me.

**Comment by christian-bromann** [comment]
Here a the zip file with all files I used including node modules: [1548.zip](https://github.com/webdriverio/webdriverio/files/426946/1548.zip)

just unpack and run `./node_modules/.bin/wdio wdio.conf.js --spec index.js`

**Comment by kajmagnus** [comment]
Hi, thanks for testing. I upgraded from wdio-mocha-framework 0.2.8 to 0.4.0 and that solved the problem.  (0.2.13 also worked OK.)

What about telling people to include their wdio-mocha-framework version too, if they use wdio, when they report errors? Not just the webdriver.io version.

Or asking them to upgrade, before reporting errors. I didn't remember I was using wdio-mocha-framework  — until I installed webdriverio from scratch in a new directory, and attempted to start it, and it said I need wdio-mocha-framework too.

(In the zip file you sent, using `browser` worked fine for me — but changing to `browserA`, didn't work, with the old 0.2.8 version of the lib.  But works with 0.4.0.)
